lib/client/src/components/posts/PostDialog.js

```diff
@@ -0,0 +1,216 @@
+      dark: "#7d968b",
+      contrastText: "#fff"
+    },
+    secondary: {
+      light: "#ff6333",
+      main: "#ff3d00",
+      dark: "#b22a00",
+      contrastText: "#fff"
+    },
+    typography: {
+      useNextVariants: true
+    },
+    form: {
+      textAlign: "center"
+    },
+    image: {
+      margin: "50px auto 20px auto",
+      width: "100px"
+    },
+    pageTitle: {
+      margin: "10px auto 10px auto"
+    },
+    textField: {
+      margin: "10px auto 10px auto"
+    },
+    button: {
+      marginTop: 30,
+      position: "relatve"
+    },
+    customError: {
+      color: "red",
+      fontSize: "0.8rem",
+      marginTop: 15
+    },
+    progress: {
+      position: "absolute"
+    }
+  }
+};
+
+class PostDialog extends Component {
+  state = {
+    open: false,
+    oldPath: "",
+    newPath: ""
+  };
+
+  handleOpen = () => {
+    // let oldPath = window.location.pathname;
+
+    // const { userHandle, screamId } = this.props;
+    // const newPath = `/users/${userHandle}/scream/${screamId}`;
+
+    // if (oldPath === newPath) oldPath = `/users/${userHandle}`;
+
+    // window.history.pushState(null, null, newPath);
+
+    this.setState({ open: true });
+    this.props.getPost(this.props.postId);
+  };
+  handleClose = () => {
+    // window.history.pushState(null, null, this.state.oldPath);
+    this.setState({ open: false });
+    // this.props.clearErrors();
+  };
+
+  render() {
+    const {
+      classes,
+      post: {
+        postId,
+        commentCount,
+        createdAt,
+        postImage,
+        username,
+        userImage,
+        comments
+      },
+      UI: { loading }
+    } = this.props;
+
+    const dialogMarkup = loading ? (
+      <div className={classes.spinnerDiv}>
+        <CircularProgress size={200} thickness={2} />
+      </div>
+    ) : (
+      <Grid container spacing={3}>
+        <Grid item xs={12}>
+          <img src={postImage} alt="Post" className={classes.postImage} />
+        </Grid>
+        <Grid item sm={7}>
+          <Typography
+            component={Link}
+            color="primary"
+            variant="h5"
+            to={`/users/${username}`}
+          >
+            @{username}
+          </Typography>
+          <hr className={classes.invisibleSeparator} />
+          <Typography variant="body2" color="textSecondary">
+            {dayjs(createdAt).format("h:mm a, MMMM DD YYYY")}
+          </Typography>
+          <hr className={classes.invisibleSeparator} />
+          <MyButton tip="Comments">
+            <ChatIcon color="primary" />
+          </MyButton>
+          <span>{commentCount} Comments</span>
+        </Grid>
+        <hr className={classes.visibleSeparator} />
+        <CommentForm postId={postId} />
+        <Comments comments={comments} />
+      </Grid>
+    );
+    return (
+      <Fragment>
+        <MyButton
+          onClick={this.handleOpen}
+          tip="Expand Post"
+          tipClassName={classes.expandButton}
+        >
```

Comment:
Instead of opening a popup, making post expand in height to show comments (with max-height and scroll enabled)